ls1intum/Artemis
# `Exam mode`: Simplify the scheduling for exam live statistics

### Checklist
#### General


#### Server

### Motivation and Context



We noticed that the current usage of the scheduled tasks causes issues on production. This implementation is updated now.

### Description


We reused the `MainInstanceMessageSendService` and the `DistributedInstanceMessageSendService` in order to schedule the exam live statistics saves after restart. This allows us to handle this task only on the main server. In addition, this should solve the production issue.

### Steps for Testing

Prerequisites:
- 1 Instructor
- One/Multiple Students
- Exam

1. Log in to Artemis
2. Create an Exam
3. Register multiple students
4. Navigate to the Exam Live Statistics dashboard
5. Enable the Exam Live Statistics
6. Participate with the students in the exam
7. Verify that actions are collected
8. Stop the Exam Live Statistics
9. Reload the Exam Live Statistics Dashboard
10. No actions should be visible (stopping the Exam Live Statistics should remove all actions from the cache)
11. Enable the Exam Live Statistics
12. Actions should be received again

### Review Progress





#### Code Review
#### Manual Tests
